include/semaphore.h

```diff
@@ -104,8 +104,16 @@ struct semholder_s
 
 struct sem_s
 {
-  volatile int32_t semcount;     /* >0 -> Num counts available */
-                                 /* <0 -> Num tasks waiting for semaphore */
+  union
+    {
+      volatile int32_t semcount;     /* >0 -> Num counts available */
+                                     /* <0 -> Num tasks waiting for semaphore */
+      volatile uint32_t mholder;     /* == NXMUTEX_NO_HOLDER -> mutex has no holder */
```

Comment:
why not keep the type as int32_t? so we can check mholder < 0 mean blocking

Reply:
Ok, decided to keep it like this for now, explanation below. I think that mutex holder is *not* an integer, but it is essentially a bit-field (but due to bitfield ordering is not standard, it is just managed via bitwise oprations/macros).

So changing that to int32 would be just confusing, I don't really like the obscure way of first setting the highest bit manually and later checking that the int has become negative. Obviously works, but again, mholder is not an integer.